AU2/plugins/custom_plugins/CompetencyPlugin.py

```diff
@@ -139,26 +170,71 @@ def on_hook_respond(self, hook: str, htmlResponse, data) -> List[HTMLComponent]:
+                            identifier=self.html_ids["Competency"],
+                            title="Extend competency?",
+                            default={},
+                            global_default=GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
+                        )
+                    ]
+                )
+            ]
+        else:
+            return []
 
     def on_event_create(self, e: Event, htmlResponse) -> List[HTMLComponent]:
-        e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = htmlResponse[self.html_ids["Competency"]]
-        return [Label("[COMPETENCY] Success!")]
+        message = []
+        competency_extension = GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Full Auto":
+            e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = htmlResponse[self.html_ids["Competency"]]
+        else:  # Create an empty competency dict when nothing was entered (due to full auto mode)
+            e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = {}
+
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Manual":
+            # auto kill competency:
+            for (killer, victim) in e.kills:
+                if not ASSASSINS_DATABASE.get(victim).is_police:
+                    if killer not in e.pluginState[self.identifier][self.plugin_state["COMPETENCY"]]:
+                        # if to avoid overwriting manual entries
+                        message.append(Label(f"[COMPETENCY] Granted {competency_extension} days competency to {killer} for a kill"))
+                        e.pluginState[self.identifier][self.plugin_state["COMPETENCY"]][killer] = competency_extension
+
+            # auto attempt competency:
+            if GENERIC_STATE_DATABASE.plugin_map.get("AttemptPlugin", False):
+                # Is this reliable? It works, but seems to depend on AttemptPlugin.on_event_create() running before this
+                # If not, how would I do this?
+                events = list(EVENTS_DATABASE.events.values())
+                events.sort(key=lambda j: j.datetime)
+                attempts_since_last_kill = {}
+
+                for event in events:
+                    for i in event.pluginState.get("AttemptPlugin", []):
+                        attempts_since_last_kill.setdefault(i, 0)
+                        attempts_since_last_kill[i] += 1
+                    for (killer, victim) in event.kills:
+                        if not ASSASSINS_DATABASE.get(victim).is_police:
+                            attempts_since_last_kill[killer] = 0
+                            # Rules unclear - should attempts be reset upon competency gain (from a kill)?

```

Comment:
Doesn't matter so long as you document it. If you're unsure, make it a config parameter.

Reply:
I'll just document it